starknet_py/net/models/transaction.py

```diff
@@ -63,7 +63,7 @@ def type(self) -> TransactionType:
         """
 
     @abstractmethod
-    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+    def calculate_hash(self, chain_id: Union[int, StarknetChainId]) -> int:
```

Comment:
I think in internal functions we can just use `chain_id: int` if pyright doesn't complain about it.